app/Http/Requests/ArticleStoreRequest.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+namespace App\Http\Requests;
+
+use App\Rules\TagExistsRule;
+use Illuminate\Foundation\Http\FormRequest;
+use Illuminate\Validation\Rule;
+
+class ArticleStoreRequest extends FormRequest
+{
+    /**
+     * Determine if the user is authorized to make this request.
+     *
+     * @return bool
+     */
+    public function authorize()
+    {
+        return true;
+    }
+
+    /**
+     * Get the validation rules that apply to the request.
+     *
+     * @return array
+     */
+    public function rules()
+    {
+        return [
+            'title' => ['required', 'string'],
+            'category_id' => ['required', 'numeric'],
+            'image' => ['nullable', 'image', 'max:2048', Rule::dimensions()->minWidth(100)->minHeight(100)->maxWidth(3000)->maxHeight(3000)],
+            'tags' => ['required', 'string', new TagExistsRule],
```

Comment:
I think laravels native exists rule can be used here https://laravel.com/docs/8.x/validation#specifying-a-custom-column-name